src/basic/process-util.c

```diff
@@ -275,21 +276,59 @@ int get_process_cmdline(pid_t pid, size_t max_length, bool comm_fallback, char *
 }
 
 void rename_process(const char name[8]) {
+        static char *mm = NULL;
+        static size_t mm_size = 0;
+        size_t l;
+        char *old_mm = NULL;
+        size_t old_mm_size = 0;
+
+        /* This is a like a poor man's setproctitle(). It changes the comm field, argv[0], and also the glibc's
+         * internally used name of the process. For the first one a limit of 16 chars applies; to the second one in
+         * many cases one of 10 (i.e. length of "/sbin/init") — however if we have CAP_SYS_RESOURCES it is unbounded;
```

Comment:
yupp, the main reason we use rename_process() is to prettyfy the "ps" output for processes PID 1 forked off but hasn't called exec() yet, maybe because it is waiting for some resource before it can do so (for example, confirm_spawn user reply, or ownership of a tty, or some NSS lookup). And then we also use it to make the PAM session supervisor process nicely visible and finally in nspawn for the stub init process. These two stay around for longer and are never replaced via execve(), and hence it's pretty nice if they show up under a nice name in "ps"